fonduer/parser/parser.py

```diff
@@ -260,6 +260,28 @@ def parse_node(node, table_info=None, figure_info=None):
                                         '='.join(x) for x in list(
                                             context_node.attrib.items())
                                     ]
+
+                                    # Extending html style attribute with the styles
+                                    # from inline style class for the element.
+                                    cur_style_index = None
+                                    for index, attr in enumerate(parts['html_attrs']):
+                                        if attr.find('style') >= 0:
+                                            cur_style_index = index
+                                            break
+                                    styles = root.find('head').find('style')
+                                    if styles is not None:
+                                        for x in list(context_node.attrib.items()):
+                                            if x[0] == 'class':
+                                                exp = r'(.' + x[1] + ')([\n\s\r]*)\{(.*?)\}'
+                                                r = re.compile(exp, re.DOTALL)
+                                                if r.search(styles.text) is not None:
+                                                    if cur_style_index is not None:
+                                                        parts['html_attrs'][cur_style_index] += r.search(styles.text).group(3)\
+                                                            .replace('\r', '').replace('\n', '').replace('\t', '')
+                                                    else:
+                                                        parts['html_attrs'] = 'style=' + r.search(styles.text).group(3)\
+                                                            .replace('\r', '').replace('\n', '').replace('\t', '')
+                                                break
```

Comment:
It's not super obvious to me what the use case for this code is. It looks to me like if an element has style attributes, then your extending the element's styles with the first style class in the head of the document that you find, and only the first style class. Is that correct? Could you explain why you needed this code?

Also, could you add a test case for this code?  This would be added to [test_parser.py](https://github.com/HazyResearch/fonduer/blob/master/tests/parser/test_parser.py).

The easiest way to do this would be to add a new, simplified HTML document into `tests/data/html/` (you can create this new html directory). Then just add a case similar to:

```py
def test_parse_style(caplog):
    """Test style tag parsing."""
    caplog.set_level(logging.INFO)
    logger = logging.getLogger(__name__)
    session = Meta.init('postgres://localhost:5432/' + ATTRIBUTE).Session()

    max_docs = 1
    docs_path = 'tests/data/html/[your new document].html'

    # Preprocessor for the Docs
    preprocessor = HTMLPreprocessor(docs_path, max_docs=max_docs)

    # Grab the document, text tuple from the preprocessor
    doc, text = next(preprocessor.generate())
    logger.info("    Text: {}".format(text))

    # Create an OmniParserUDF
    omni_udf = OmniParserUDF(
        True,           # structural
        [],             # blacklist, empty so that style is not blacklisted
        ["span", "br"], # flatten
        '',             # flatten delim
        True,           # lingual
        True,           # strip
        [],             # replace
        True,           # tabular
        True,           # visual
        pdf_path,       # pdf path
        Spacy())        # lingual parser

    # Grab the phrases parsed by the OmniParser
    phrases = list(omni_udf.parse_structure(doc, text))

    # Add your assertions
```

Reply:
If an html element `<elm class='.s2'>` has a class attribute `here .s2` then this code will find that class `s2` in `<style>`  and append/assign the `styles` defined in the `s2` class to  style attribute in `parts['html_attrs']`

Reply:
Will add test case and check in